### ISSUE TITLE
LineSeries.from_csv Issues

### ISSUE THREAD

**Issue by thesteed** [highcharts-for-python/highcharts-core#32]
**Describe the bug**

Several issues:
1) `checkers.is_file_on_filesystem(as_string_or_file)` in base.py I believe should be:
`checkers.is_on_filesystem(as_string_or_file)`
Assuming it's this library: https://github.com/insightindustry/validator-collection/tree/master

After fixing the above issue:

2) It appears at least the `wrapper_character` parameter is ignored, possibly others

3) When passing in a well formatted CSV string, (digestible via pythons CSV when matching the `highcharts` dialect specified in utility_functions.py) , the utility_functions.parse_csv fails to parse it.

**To Reproduce**
Run:
```
from highcharts_core.options.series.area import LineSeries

series1 = LineSeries.from_csv("'Date','HeadCount'\r\n'01/01/2023','2'\r\n'01/02/2023','4'\r\n'01/03/2023','8'", property_column_map={'x': 'Date', 'y': 'HeadCount', 'id': 'Date'})
```
Which yields:

```
columns: ['Date']
csv_records: [{'Date': '', None: ['']}, {'Date': 'HeadCount'}, {'Date': '01/01/2023'}, {'Date': '', None: ['']}, {'Date': '2'}, {'Date': '01/02/2023'}, {'Date': '', None: ['']}, {'Date': '4'}, {'Date': '01/03/2023'}, {'Date': '', None: ['']}, {'Date': '8'}]
```
Which was printed via putting
`print(f'columns: {columns}\ncsv_records: {csv_records}')`
In `base.py` immediately after:
```
   columns, csv_records = utility_functions.parse_csv(
            as_str,
            has_header_row = has_header_row,
            delimiter = delimiter,
            null_text = null_text,
            wrapper_character = wrapper_character,
            line_terminator = line_terminator,
            wrap_all_strings = False,
            double_wrapper_character_when_nested = False,
            escape_character = "\\"
        )
```

I've attempted many variations of the CSV string, all seem to yield similar results.

**Expected behavior**

I would expect it to parse this CSV string into a series, where each record in the series has an 'x' value populated from the 'Date' column of the CSV and a 'y' value populated from the 'HeadCount' column.

**Your Environment:**
 - OS: Ubuntu 20.04 LTS
 - Python Version: 3.10.6
 - Highcharts Python: 1.0.0

**Additional context**

**Comment by hcpchris** [comment]
@thesteed Thanks for reporting the bug! Let me look into it - I'll hopefully have it patched shortly, though in practice the patch might not get deployed until sometime over the weekend.

**Comment by thesteed** [comment]
@hcpchris Sounds good!

**Comment by hcpchris** [comment]
Ah - found the issue, and added some precise unit tests to catch it going forward. The PR with the fix ( #33 ) is going through our CI/CD release checks now, so hopefully **v.1.0.1** will be released this evening (probably around 9 or 10pm Eastern, if not a little earlier). 

This issue will get auto-closed when the PR gets merged into the ``master`` branch (which should happen about 1 minute before it gets released to PyPi). Thanks again for bringing this issue to our attention! If you run into any other issues, please let us know.
